libr/core/canal.c

```diff
@@ -2214,11 +2211,8 @@ R_API int r_core_anal_fcn_list(RCore *core, const char *input, const char *rad)
 		fcn_list_default (core, fcns, false);
 		break;
 	}
-	//make sure you don't free core->anal->fcns
-	if (input && core->anal->fcns != fcns) {
-		// The list does not own the its members, so don't purge.
-		free (fcns);
-	}
+
+	free (fcns);
```

Comment:
this leaks. you cant free" a list.. it must use r_list_free or r_list_destroy.. unset the->free ptr if you just wanna free the container and not the content